build/Dockerfile

```diff
@@ -94,22 +106,18 @@ RUN dnf install -y python3.11 git && \
     ln -s /usr/bin/python3.11 /bin/python && \
     python -m ensurepip --upgrade
 
-RUN mkdir /app
-
 WORKDIR /tmp
-RUN python -m pip install --upgrade pip
-
-# TODO Move to installing wheel once we have proper releases setup instead of cloning the repo
-RUN git clone https://github.com/foundation-model-stack/fms-hf-tuning.git && \
-    cd fms-hf-tuning && \
-    python -m pip install ".[dev]" && \
-    python -m pip install ".[flash-attn]" && \
-    python -m pip install ".[aim]" && \
-    python -m pip install -U datasets
+COPY --from=build /tmp/*.whl /tmp/bdist_name /tmp/
+# ami and flash-attn depends on dev
+RUN --mount=type=cache,target=/root/.cache/pip \
+    python -m pip install "$(head bdist_name)[dev]" && \
```

Comment:
@tedhtchang if we install all the packages as root, and when we run the docker image as user, will there be any permission problems?

Reply:
The user should be in root group. Since all files by default are own by root group so the user should have access. I think it's ok.

Reply:
for security reasons its best to run the final image as a non-root user. this does limit access but I don't think there should be permissions issues related to running the wheel.